docs/GETTING_STARTED.md

```diff
@@ -29,6 +29,9 @@ https://github.com/awslabs/soci-snapshotter.git`
 Often fixed with `sudo yum install zlib-devel`
 - having the proper go version installed, as of this document please use 1.18+
 
+**fuse** - confirm FUSE is installed for mounting userspace filesystems with `fusermount -V`
+(optional, enables mounting without root access)
```

Comment:
I guess. This does need to be documented. And I guess until we have a user guide, this is a place to do it. But a *Getting Started* guide should be low-friction. Every added step, every added bit of information, every chance for something to go wrong translates to a non-zero number of users who will walk away before successfully using SOCI. Since the examples below all use `sudo`, I'm still left wondering why I'm being told about this now.

When this *is* added to the [non-existent] user guide, it should be completely documented, since the code is [a bit opaque](https://github.com/awslabs/soci-snapshotter/blob/106fdbb1ea903aa2535cb0395f8764d72892e796/fs/fs.go#L509) about the implications. When I look at the effect that this has in the [fuse client code](https://github.com/hanwen/go-fuse/blob/master/fuse/mount_linux.go#L137-L138), it makes me want to *uninstall* the fuse package. I trust *mount()*. I don't trust "magic".

Reply:
Is this because fusermount is typically setuid'd?

https://github.com/libfuse/libfuse#security-implications

> The fusermount3 program is installed setuid root. This is done to allow normal users to mount their own filesystem implementations.

As libfuse's README said,

> FUSE (Filesystem in Userspace) is an interface for userspace programs to export a filesystem to the Linux kernel. The FUSE project consists of two components: the fuse kernel module (maintained in the regular kernel repositories) and the libfuse userspace library (maintained in this repository). libfuse provides the reference implementation for communicating with the FUSE kernel module.

We should separate the kernel module and the libfuse userspace library along with userspace binaries. The former is must, the latter is optional.

Then regarding the optional dependency, can you confirm that the snapshotter works without root permissions if there is fusermount? If it works end-to-end, mentioning that is valuable for customers, if not, we don't have to ask customers to have fusermount.

Reply:
I was really hoping with this PR to just make the existing documentation and reported behavior more clear. I can open an issue to do further research if necessary.

Reply:
https://github.com/awslabs/soci-snapshotter/issues/389